iphone/Classes/TiApp.m

```diff
@@ -607,11 +607,26 @@ - (void)application:(UIApplication *)application handleEventsForBackgroundURLSes
 
 -(void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTask *)downloadTask didFinishDownloadingToURL:(NSURL *)location
 {
-    //FunctionName();
-    TiBlob * downloadedFile =[[[TiBlob alloc] initWithData:[NSData dataWithContentsOfURL:location] mimetype:[Mimetypes mimeTypeForExtension:[location absoluteString]]] autorelease];
-    NSMutableDictionary *dict = [NSMutableDictionary dictionaryWithObjectsAndKeys: [NSNumber numberWithUnsignedInteger:downloadTask.taskIdentifier ],@"taskIdentifier",downloadedFile,@"data", nil];
-    [[NSNotificationCenter defaultCenter] postNotificationName:kTiURLDownloadFinished object:self userInfo:dict];
-    
+	//FunctionName();
+	//copy downloaded file from location to tempFile (in NSTemporaryDirectory), because file in location will be removed after delegate completes
+	NSError *error;
+	NSFileManager *fileManager = [NSFileManager defaultManager];
+	NSString *destinationFilename = downloadTask.originalRequest.URL.lastPathComponent;
```

Comment:
This is going to be a problem. What if the original URL in the request has no path component
`http://www.myfileserved.com/`